### ISSUE TITLE
PM->TypeError:Cannot read properties of undefined (reading 'children')

### ISSUE THREAD

**Issue by kenjibailly** [vrtmrz/livesync-classroom#1]
I had it working for a second, but it keeps going in error even after reverting the changes on the config.json file.
This error comes from the "path" variable I'm pretty sure, although I tried to change it and now it keeps on giving me the same error:
`PM->TypeError:Cannot read properties of undefined (reading 'children')`

I had it working with the following config:

```json
{
    "private": {
        "uri": "https://url.com/mindglowing",
        "auth": {
            "username": "",
            "password": "",
            "passphrase": ""
        },
        "path": "Atrilluh/"
    },
    "shared": {
        "uri": "https://url.com/atrilluh",
        "auth": {
            "username": "",
            "password": "",
            "passphrase": ""
        },
        "path": "Atrilluh/"
    }
}
```

What I wanted to try is to change the path of the private vault to something like "Obsidian Vault/Collabs/Atrilluh/", but when doing that change and going back it now keeps on saying that error.

Ofc I stopped the script and reran it a couple times. 
Not sure what I'm doing wrong.

Also big thanks for creating this project!!! 
Everything is making good sense of what it's doing, except for the path variable that bugs on me.

I also checked if the hash file was in the DB of the source to where it should sync and it is, so it should be able to find it.

---

EDIT:

Fixed the problem:
I removed the replication from collab > private and apparently that caused this issue.
So a replication from collab > private configured in the database is required to run the script, otherwise it won't work.

The only problem I see here is say I wanted to create a 3rd DB where the user of the collab DB uses his own private vault.
So it would be looking like this:

3 databases: 
- mindglowing (my private vault)
- atrilluh (collab user)
- atrilluh-private (collab user private db vault)

Private Vault:
- Obsidian Vault/all private files here of the mindglowing db
- Collab/Atrilluh/all collab files of atrilluh database here

Private Vault of Collab User:
- Obsidian Vault/all private files here of the new user db, let's call it atrilluh-private
- Collab/Atrilluh/all collab files of atrilluh database here

Perhaps this is already possible, but I'm doing things wrong?
Please let me know, and thank you so much!!!! You're the best!

**Comment by vrtmrz** [comment]
@kenjibailly 
Multiple vault synchronization is implemented! And I wrote the readme! (Yes, I'm terrible at this.)
And thank you for the pull requests about docker. I'm not familiar with docker, it very helps me!
Could you try this?

**Comment by kenjibailly** [comment]
> @kenjibailly Multiple vault synchronization is implemented! And I wrote the readme! (Yes, I'm terrible at this.) And thank you for the pull requests about docker. I'm not familiar with docker, it very helps me! Could you try this?

Thank you so much!!! I tried to do the Obsidian Vault/Collab/Atrilluh in the private vault as I mentioned earlier and it's working perfectly even when the target folder has not the same name!
I also removed the replication I talked about earlier all still working fine. 
I tried a few things, so basically I have 2 configurations right now and 3 databases. 2 private vaults that sync with a shared collab folder and no errors, bugs or whatever along the road. 

You're fire man!!!! 
And you're most welcome for the docker config, I only use docker applications on my system since things always seem to break with me after time and I've got my proper amount of reinstalls already :p I guess I'm too much of a linux noob to fix these difficult issues sometimes when 2 applications are getting totally confused or when my ethernet adapter is just suddenly gone and unable to retrieve. 

I def can't do what you did in this project, it's also my first pull request on github ever :D feels good hehe

**Comment by Manel9999** [comment]
This is what is showing me: Error handling response: TypeError: Cannot read properties of undefined (reading 'vault') at <anonymous>:2:44. Any thoughts? I am ejecuting this on the consola for restoring info

**Comment by vrtmrz** [comment]
@Manel9999 
Hi! Thank you for asking me!
It's a similar message but would be caused by another reason. Could you make an issue about this? 
(I'm very glad to hear the information about when does it be shown?)

**Comment by Manel9999** [comment]
Hi!! Basically what happens is: I created an account few months ago where I had some tokens ( IMPORTANT: I did not created any seed phrase for that account while creating it, it only required to set a PWD). One week ago, I logged in with another computer ( I entered the seed I thought I created for the account but accidentally a new account was created with no tokens there). My tokens were GMX and I was using metamask public key to connect to gmx.io. Now, I enter on gmx.io, conect the wallet ( the wrong one) and no GMX tokens appear...

What I am trying to do is obtain my seed phrase for the first and main wallet so I am executing this querie on the console in order to use the vault encryptor and find the seed phrase but two things happen:
chrome.storage.local.get('data', result => {
 var vault = result.data.KeyringController.vault
 console.log(vault)
}) 

1- If I am logged in to the wrong account ( the second I accidentally created) while running the querie on the console, some data appears, but when I use this data on the vault encryptor, lots of numbers come up instead of the 12 seed phrase words, therefore I can't log into the correct account. I think is due to I am already logged in the wrong wallet perhaps
2- If I amb not logged in any meta wallet ( this error comes up): TypeError: Cannot read properties of undefined (reading 'vault') at :2:44

To sum this up, I need to find my seed phrase for the first account ( I have the public key) but there is no way to lkog in meta wallet with this key...
